src/parsers/containers/isobmff/utils.ts

```diff
@@ -327,83 +303,115 @@ function getMDHDTimescale(buffer : Uint8Array) : number {
- * object containing two properties:
- *   - systemId {string}: The uuid code. Should only contain 32 hexadecimal
- *     numbers and hyphens
- *   - privateData {Uint8Array} private data associated.
+ * Creates a PSSH box with the given systemId and data.
+ * @param {Array.<Object>} psshInfo
  * @returns {Uint8Array}
  */
-function createPssh({ systemId, privateData } : IISOBMFFKeySystem) : Uint8Array {
+function createPssh({ systemId, privateData } : IISOBMFFPSSHInfo) : Uint8Array {
   const _systemId = systemId.replace(/-/g, "");
 
   assert(_systemId.length === 32);
-  return createBox("pssh", concat(
-    4, // 4 initial zeroed bytes
-    hexToBytes(_systemId),
-    itobe4(privateData.length),
-    privateData
-  ));
+  return createBox("pssh", concat(4, // 4 initial zeroed bytes
+                                  hexToBytes(_systemId),
+                                  itobe4(privateData.length),
+                                  privateData));
 }
 
 /**
  * Update ISOBMFF given to add a "pssh" box in the "moov" box for every content
  * protection in the pssList array given.
  * @param {Uint8Array} buf - the ISOBMFF file
- * @param {Array.<Object>} pssList - The content protections under the form of
- * objects containing two properties:
- *   - systemId {string}: The uuid code. Should only contain 32 hexadecimal
- *     numbers and hyphens
- *   - privateData {Uint8Array} private data associated.
+ * @param {Array.<Object>} pssList
  * @returns {Uint8Array} - The new ISOBMFF generated.
  */
-function patchPssh(buf : Uint8Array, pssList : IISOBMFFKeySystem[]) : Uint8Array {
+function patchPssh(buf : Uint8Array, pssList : IISOBMFFPSSHInfo[]) : Uint8Array {
   if (pssList == null || pssList.length === 0) {
     return buf;
   }
 
-  const pos = findBox(buf, 0x6D6F6F76 /* = "moov" */);
-  if (pos === -1) {
+  const moovOffsets = getBoxOffsets(buf, 0x6D6F6F76 /* = "moov" */);
+  if (moovOffsets === null) {
     return buf;
   }
 
-  const size = be4toi(buf, pos); // size of the "moov" box
-  const moov = buf.subarray(pos, pos + size);
-
+  const moov = buf.subarray(moovOffsets[0], moovOffsets[2]);
   const moovArr = [moov];
   for (let i = 0; i < pssList.length; i++) {
     moovArr.push(createPssh(pssList[i]));
   }
+  const newmoov = updateBoxLength(concat(...moovArr));
 
-  const newmoov = concat(...moovArr);
-  newmoov.set(itobe4(newmoov.length), 0); // overwrite "moov" length
+  return concat(buf.subarray(0, moovOffsets[0]),
+                newmoov,
+                buf.subarray(moovOffsets[2]));
+}
 
-  return concat(
-    buf.subarray(0, pos),
-    newmoov,
-    buf.subarray(pos + size)
-  );
+/**
+ * Returns a new version of the given box with the size updated
+ * so it reflects its actual size.
+ *
+ * You can use this function after modifying a ISOBMFF box so its size is
+ * updated.
+ *
+ * /!\ Please consider that this function might mutate the given Uint8Array
```

Comment:
Why not always creating a new one ?

Reply:
It could be clearer when using the func

Reply:
As discused, this is mainly for performance consideration.

Mutating in place is what we will do in 99.9999...% of cases.
Copying the whole buffer has a cost (even more if we're talking about big boxes like moov or even mdat) which does not need to be incurred in any real case.